spec/metrics/3scale_backend_calls_spec.lua

```diff
@@ -0,0 +1,73 @@
+describe('backend calls metrics', function()
+  describe('report', function()
+    local threescale_backend_calls_metrics
+    local test_counter = { inc = function() end }
+
+    before_each(function()
+      stub(test_counter, 'inc')
+      local Prometheus = require('apicast.prometheus')
+      getmetatable(Prometheus).__call = function(_, type)
```

Comment:
Doing this dance in tests is so ugly and error-prone, that I think this could be an excellent time to fix this interface.

Reply:
Agreed. I thought the same because it's the second time we need to write something like this.
I was thinking about whether to modify it in this PR or the next one.

Reply:
I'm good either way.

Reply:
Let's address this one separately because it will change many things not directly related to this PR.